### ISSUE TITLE
Shape creation via side effect does not allow for named objects

### ISSUE THREAD

**Issue by thomastjeffery** [cbiffle/ruckus#5]
Take for example this code:

```racket
#lang ruckus

(let ([ball (sphere 200)]
      [box (cube 300)])
  (difference box ball))
```

This crashes, since `difference` is given `'()`.

Without the call to `difference`, we would have this confusing program:

```racket
#lang ruckus

(let ([ball (sphere 200)]
      [box (cube 300)])
  ball)
```

Instead of showing us only the sphere, this program would show us *both* the ball and sphere, and happily ignore the fact that the `let` form evaluates to `'()`.

&nbsp;

This isn't simply confusing behavior: it's leaves no method for naming objects.

It's a bit like a strawman version of racket where all strings evaluated to `'()`, while having the side effect of printing to the console. Sure, "Hello, world!" would be simpler, but at great cost.

**Comment by nucea-skulero** [comment]
> This isn't simply confusing behavior: it's leaves no method for naming objects.

I have to agree--the current DSL might at least include a ```letshape``` with the same syntax/semantics as ```let```.  I've done a fair amount of modeling with Libfive in Scheme/Common Lisp, and used this feature extensively.  

Alternately, the current design could just use functions that return AST tree nodes, which would then allow using plain old ```let```.  I'm not sure why macros would have to be used.  Maybe I'm missing something.

[As an aside, it's disappointing that Ruckus didn't get more attention.  It is a great effort at F-rep modeling, elegant and flexible.  Treating the F-rep modeling process much like a compiler is a quite nice approach.  As well as supporting different backends, it does seem to facilitate adding alternate language front-ends like C, Julia, Python, etc.]
